### DISCUSSION TITLE
Useful development tools and techniques

### DISCUSSION THREAD

**Discussion by brylie** [CiviWiki/OpenCiviWiki#1093] [General]
Use this topic to discuss useful development tools and techniques.

**Comment by brylie** [comment]
I use the open-source VSCode IDE for development because it includes or supports

- Git/[GitHub integration](https://vscode.github.com/)
- live pair-programming via [Live Share](https://code.visualstudio.com/learn/collaboration/live-share)
- [Python integration](https://code.visualstudio.com/docs/languages/python) and virtual environment support
- automatic [code linting](https://code.visualstudio.com/docs/python/linting) and formatting
- code auto-completion via [Intellisense](https://code.visualstudio.com/docs/editor/intellisense)
- quick [code navigation](https://code.visualstudio.com/Docs/editor/editingevolved)
- lots of great extensions with [integrated extension management](https://code.visualstudio.com/docs/editor/extension-marketplace)

**Reply by ruben216** [reply]
Hello Brylie,
I'm a Data Science student, and I'd like to contribute with this project, in any way.
Today my MBA is in R language, but I'm studying Python, web development , etc. 
Please, could you share any tips(and tool)  to debug the issues created for this project?
Thanks a lot!

**Reply by brylie** [reply]
Aside from what I shared above and the built-in VS Code debugger, I can recommend the [Django Debug Toolbar](https://django-debug-toolbar.readthedocs.io/en/latest/).

**Comment by Green-Abe** [comment]
I use terminal and text editor (Ubuntu)
